src/Type.cpp

```diff
@@ -307,7 +322,10 @@ Type::Type(const Type* t) :
     ptr_type(t),
     used(false),
     printed(false),
-    packed_(false)
+    packed_(false),
+	simple_type(MAX_SIMPLE_TYPES),               // not a valid simple type
+	has_assign_ops_(false),
+    has_implicit_nontrivial_assign_ops_(false)
```

Comment:
Use the following order? 

    ptr_type(t),
    simple_type(MAX_SIMPLE_TYPES),               // not a valid simple type
    used(false),
    printed(false),
    packed_(false),
    has_assign_ops_(false),
    has_implicit_nontrivial_assign_ops_(false)